addons/token-exchange/token_exchanger_agent.go

```diff
@@ -153,22 +161,65 @@ func (c *tokenExchangeAgentController) sync(ctx context.Context, syncCtx factory
 		return nil
 	}
 
-	se, err := c.spokeSecretLister.Secrets(namespace).Get(name)
-	switch {
-	case errors.IsNotFound(err):
+	// Sync secret from hub to spoke
+	err = c.syncToSpoke(name, namespace)
+	if err != nil {
+		return fmt.Errorf("failed to sync hub secrets. %v", err)
+	}
+
+	return nil
+}
+
+// syncToSpoke syncs a secret from hub to spoke cluster
+func (c *tokenExchangeAgentController) syncToSpoke(name, namespace string) error {
+	secret, err := getSecret(c.hubSecretLister, name, namespace)
+	if err != nil {
+		return fmt.Errorf("failed to get the sercet %q in namespace %q in hub", name, namespace)
+	}
+
+	if secret == nil {
 		return nil
-	case err != nil:
-		return err
 	}
 
-	secret := &corev1.Secret{
+	if secret.GetLabels()[secretutils.SecretLabelTypeKey] != string(secretutils.DestinationLabel) {
+		klog.Infof("secret %q in namespace %q is not a green secret. Skip syncing with the spoke cluster", secret.Name, namespace)
+		return nil
+	}
+
+	toNamespace := string(secret.Data["namespace"])
+	if toNamespace == "" {
+		return fmt.Errorf("missing storageCluster namespace info in secret %q in namespace %q", secret.Name, namespace)
+	}
+
+	newSecret := &corev1.Secret{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      se.Name, // TODO: Handle naming conflict
-			Namespace: "default",
+			Name:      secret.Name,
+			Namespace: toNamespace,
+			Labels:    map[string]string{CreatedByLabelKey: CreatedByLabelValue},
 		},
-		Data: se.Data,
+		Data: map[string][]byte{"secret-data": secret.Data["secret-data"]},
```

Comment:
Can we move these string literals (in the map keys, like "secret-data" / "namespace" etc) to the `common` constants?